compiler/damlc/daml-stdlib-src/DA/Date.daml

```diff
@@ -38,6 +40,19 @@ deriving instance Show Month
 deriving instance Enum Month
 deriving instance Bounded Month
 
+instance Enum Date where
+    toEnum n = daysSinceEpochToDate (n - 719162)
+    fromEnum d = 719162 + dateToDaysSinceEpoch d
+    enumFrom     x   = enumFromTo     x maxBound
+    enumFromThen x y = enumFromThenTo x y bound
+        where
+            bound | fromEnum y >= fromEnum x = maxBound
```

Comment:
We can just compare the dates without going through `fromEnum`:
```suggestion
            bound | y >= x = maxBound
```
I would also just inline the definition of `bound` using an `if ... then ... else ...`, but that's a matter of personal taste.

Reply:
The definition here is copied from the docs. Happy to change it but I think there’s also value in sticking to that.

Reply:
I assume the docs have to be that general because they can't assume that `a` is `Ord`. Here, we know that `Data` is `Ord`.